### DISCUSSION TITLE
How to use brew with two user accounts?

### DISCUSSION THREAD

**Discussion by Shpota** [Homebrew/discussions#66] [Everyday usage]
I have two admin accounts on the same machine. An old one and a newly created one. Now I want to be able to use brew with the new account as well.

Running `brew update` leads to the following error:
```
Error: Can't create update lock in /usr/local/var/homebrew/locks!
Fix permissions by running:
  sudo chown -R $(whoami) /usr/local/var/homebrew
```
Executing the suggested command makes it possible to run `brew update` but it still prints errors with missing permissions. Installing packages also doesn't work. It suggests executing other commands and changing permissions. But I am not sure I won't break anything if I keep executing the suggested commands. 

Unfortunately, I couldn't find a solution in the documentation.

**Question:** How to use `brew` with several user accounts? Shall I have two brew installations (one per account)? Is there a guide on this (considering I am not an expert in this area)?

**Comment by SMillerDev** [comment]
https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

> If you need to run Homebrew in a multi-user environment, consider creating a separate user account especially for use of Homebrew.

**Comment by gromgit** [comment]
https://discourse.brew.sh/t/how-to-allow-a-non-admin-user-download-and-install-homebrew/8089/2

**Comment by iamtodor** [comment]
I find this answer https://stackoverflow.com/a/59691631/5151861 helpful. Is it a proper way to use it?

**Reply by gromgit** [reply]
Well, it gets at least two things wrong:
1. It assumes that all user accounts on Catalina and after use Zsh as the login shell. That's only true for _freshly-created accounts_; existing accounts retain their original login shell from the time of creation (or last modification), all the way through multiple OS upgrades and migrations to new hardware. Blindly adding a "change `PATH`" command to `~/.zprofile` does nothing useful at all if your login shell is `bash`, or `tcsh`...or `fish`.
2. `eval "$(/path/to/brew shellenv)"` is the canonical way of setting up a specific Homebrew instance's environment for the user, and is documented in [the section of the install docs actually used by the answerer](https://docs.brew.sh/Installation#untar-anywhere). Merely changing `PATH` falls short in at least three ways:
   1. it doesn't make things like man pages and info docs available to the end-user
   2. `export PATH="$HOME/homebrew/bin:$PATH"` is likely a syntax error in non-Bourne shells, while `brew shellenv` is smart enough to emit the correct instructions for the shell in use
   3. `brew shellenv` may add additional settings as and when they become necessary
